plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -492,7 +496,7 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
                 .addAll(getColumnInvariants(metadataEntry).values()) // The internal logic for column invariants in Delta Lake is same as check constraints
                 .build();
         List<ColumnMetadata> columns = getColumns(metadataEntry).stream()
-                .map(column -> getColumnMetadata(column, columnComments.get(column.getName()), columnsNullability.getOrDefault(column.getName(), true), columnGenerations.get(column.getName())))
+                .map(column -> getColumnMetadata(column, columnComments.get(column.getBaseColumnName()), columnsNullability.getOrDefault(column.getBaseColumnName(), true), columnGenerations.get(column.getBaseColumnName())))
```

Comment:
- consult `columnComments` `columnsNullability` and `columnGenerations` only for base columns
- `getColumns(metadataEntry)` is correct only if table handle doesn't have projections; if the table handle has dereference columns, we should create metadata for these

fortunately, the `getTableMetadata` is used only in context where there cannot be pre-projected columns
so add a check at the top of the method, an no other chhanges needed here

Reply:
When you say table handle projections, do you mean by `DeltaLakeTableHandle#projectedColumns`?

I could see that in `getTableMetadata` calls, `DeltaLakeTableHandle#projectedColumns` has dereference columns. The call to `getTableMetadata` comes from `ConnectorMetadata#getTableSchema`.

Table Schema:
```
 CREATE TABLE delta_lake.dereference_pushdown.testdelta (
    a integer,
    r1 ROW(r2 ROW(a integer, b integer), c integer) COMMENT 'column-comment',
    d integer COMMENT 'column-comment'
 );

SELECT r1.r2.b, d, r1.r2.a from testdelta where a = 1;
```

Reply:
> I could see that in `getTableMetadata` calls, `DeltaLakeTableHandle#projectedColumns` has dereference columns. The call to `getTableMetadata` comes from `ConnectorMetadata#getTableSchema`.

you're right

Reply:
https://github.com/trinodb/trino/pull/17269